### ISSUE TITLE
Error: Plugin did not respond

### ISSUE THREAD

**Issue by vinitkapoor** [oracle/terraform-provider-oci#1629]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

$ terraform -v
Terraform v1.1.4
on darwin_amd64
+ provider registry.terraform.io/oracle/oci v4.71.0

Initializing provider plugins...
- Finding oracle/oci versions matching "4.82.0"...
- Installing oracle/oci v4.82.0...
- Installed oracle/oci v4.82.0 (signed by a HashiCorp partner, key ID 1533A49284137CEB)

### Affected Resource(s)

oci_core_ipsec_connection_tunnel_management

### Terraform Configuration Files

```hcl
resource "oci_core_ipsec_connection_tunnel_management" "oci_ipsec_tunnel-b" {
  #Required
  for_each  = var.ipsec_connection_details
  ipsec_id  = oci_core_ipsec.oci_ip_sec[each.key].id
  tunnel_id = data.oci_core_ipsec_connection_tunnels.oci_tunnels[each.key].ip_sec_connection_tunnels[1].id
  routing   = each.value.routing

  #Optional
  bgp_session_info {
    #Optional
    customer_bgp_asn      = each.value.tunnel_two.customer_bgp_asn
    customer_interface_ip = each.value.tunnel_two.customer_interface_ip
    oracle_interface_ip   = each.value.tunnel_two.oracle_interface_ip
  }
  display_name  = "TUNNEL-2"
  shared_secret = var.shared_secret
  ike_version   = each.value.ike_version
}
```

### Debug Output

╷
│ Error: Plugin did not respond
│ 
│   with oci_core_ipsec_connection_tunnel_management.oci_ipsec_tunnel-b["connection_2"],
│   on main.tf line 64, in resource "oci_core_ipsec_connection_tunnel_management" "oci_ipsec_tunnel-b":
│   64: resource "oci_core_ipsec_connection_tunnel_management" "oci_ipsec_tunnel-b" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-oci_v4.82.0 plugin:

panic: interface conversion: interface {} is string, not int

goroutine 240 [running]:
terraform-provider-oci/internal/service/core.(*CoreIpSecConnectionTunnelManagementResourceCrud).Update(0xc00293c600)
        terraform-provider-oci/internal/service/core/core_ipsec_connection_tunnel_management_resource.go:500 +0x3338
terraform-provider-oci/internal/tfresource.UpdateResource({0x66a0c10, 0xc000ad3100}, {0x65b14f0, 0xc00293c600})
        terraform-provider-oci/internal/tfresource/crud_helpers.go:297 +0x139
terraform-provider-oci/internal/service/core.updateCoreIpSecConnectionTunnelManagement(0xc000ad3100, {0x5e46580, 0xc002d2e408})
        terraform-provider-oci/internal/service/core/core_ipsec_connection_tunnel_management_resource.go:317 +0xe5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00293c5d0, {0x6687c28, 0xc0024e4700}, 0x24, {0x5e46580, 0xc002d2e408})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:352 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0017fdce0, {0x6687c28, 0xc0024e4700}, 0xc00036aa80, 0xc0028bd2c0, {0x5e46580, 0xc002d2e408})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:464 +0x6ba
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000710228, {0x6687c28, 0xc0024e4700}, 0xc001de38b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x9aa
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0004d0cc0, {0x6687cd0, 0xc0024e1cb0}, 0xc001de3450)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0x6c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5b89660, 0xc0004d0cc0}, {0x6687cd0, 0xc0024e1cb0}, 0xc0029b5620, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00070ec40, {0x66c18d0, 0xc0005fe000}, 0xc000587200, 0xc001123c20, 0x91a68a0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc00070ec40, {0x66c18d0, 0xc0005fe000}, 0xc000587200, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x294

Error: The terraform-provider-oci_v4.82.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

ERRO[0065] 1 error occurred:
        * exit status 1

### Expected Behavior

IPSec connection should have been created.

### Actual Behavior

Terraform plugin error

### Steps to Reproduce

1. terraform init.
2. terrafrom plan
3. terraform apply

### Important Factoids

None

### References

None

**Comment by ravinitp** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by iejones** [comment]
I believe this is the same bug reported in[ #1563 ](https://github.com/oracle/terraform-provider-oci/issues/1563) which was fixed in July 2022 with release 4.85.0
